### PR TITLE
Improve default seed generation in Random module

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Improve the default seed generated for a `randomStream` by including the following pieces of information:
* process ID
* task ID
* locale ID
* current time
* bits from `/dev/urandom` (if available)

This extra information will significantly reduce the likelihood that two `randomStream`'s generated around the same time will have the same seed value.

Resolves: https://github.com/chapel-lang/chapel/issues/24178

- [x] paratest

### CODE REVIEW

File: modules/standard/Random.chpl [hunk 1]
```diff
@@ -121,11 +121,22 @@ module Random {
...
-    const oddseed = if seed % 2 == 0 then seed + 1 else seed;
-    return oddseed;
+  private proc randomishSeed(): int {
+    import Time, IO, CTypes;
+    extern proc chpl_task_getId(): chpl_taskID_t;
+    extern proc getpid(): CTypes.c_int;
+
+    const sWho = chpl_task_getId().hash():int,
+          sWhat = getpid():int,
+          sWhen = (Time.timeSinceEpoch().totalSeconds()*1_000_000):int,
```
Comment (mppf) [inline modules/standard/Random.chpl:131]: seems like you should hash this value as well as the pid value

File: modules/standard/Random.chpl [hunk 2]
```diff
@@ -121,11 +121,22 @@ module Random {
...
+          sWhere = here.hash():int;
+
+    try {
+      var urand = IO.openReader("/dev/urandom");
```
Comment (mppf) [inline modules/standard/Random.chpl:135]: `/dev/urandom` won't exist on Mac OS X, so the error handling is important here

Comment (jeremiah-corrado) [reply modules/standard/Random.chpl:135]: Are you saying that there needs to be additional error handling other than the try-catch block?

Comment (jeremiah-corrado) [reply modules/standard/Random.chpl]: Good point, I've refactored it

**Review by mppf — COMMENTED** [review]
`randomishSeed` looks good to me.

I am more uncertain about whether or not we are ready to remove `@unstable` from several RNG functions with this change. The change itself in the PR looks fine if you are confident that we are ready to do that.

**Comment by jeremiah-corrado** [comment]
Great thanks!

> I am more uncertain about whether or not we are ready to remove @unstable from several RNG functions with this change.

I believe we decided that those could be removed once we were confident that default seed generation was more robust (essentially once https://github.com/chapel-lang/chapel/issues/24178 is resolved). Did you have a different impression from the [last design discussion](https://github.com/Cray/chapel-private/issues/5604#issuecomment-1883773167) about why the unstable warnings are there?

**Comment by jeremiah-corrado** [comment]
After some offline discussion, we've decided to keep the routines in question unstable for the time being and remove unstable warnings in a separate PR.

**Comment by bradcray** [comment]
Nice!

**Pull request merged**